query_reformulation.py

```diff
@@ -3,29 +3,69 @@
 import os
 import json
 import requests
+import logging
+
+logger = logging.getLogger('goat_nlp.query_reformulation')
 
 
 def fetch_related_taxons(query: str):
+    """
+    Fetches related taxons for a given query.
+
+    Args:
+        query (str): The query for which related taxons need to be fetched.
+
+    Returns:
+        dict: A dictionary mapping entities to their corresponding taxons.
+
+    Raises:
+        Exception: If an error occurs while making the API call to retrieve
+            taxons.
+
+    Example:
+        >>> query = "find the number of assemblies for bat"
+        >>> fetch_related_taxons(query)
+        {'bat': ['Chiroptera', 'bat']}
+
+    """
     entity_taxon_map = {}
-    for i in range(int(os.environ["RETRY_COUNT"])):
+    for i in range(int(os.environ.get("RETRY_COUNT", 3))):
```

Comment:
sorry to change my previous suggestion, but I just remembered `os.getenv` for a slightly neater way to do this